server/validate.go

```diff
@@ -38,8 +38,32 @@ func validateGetInclusionProofByHashRequest(req *trillian.GetInclusionProofByHas
 	if req.TreeSize <= 0 {
 		return status.Errorf(codes.InvalidArgument, "GetInclusionProofByHashRequest.TreeSize: %v, want > 0", req.TreeSize)
 	}
+	if !isLeafHashValid(req.LeafHash) {
```

Comment:
Maybe in-line this `len() > 0` test since the error message is quite specifically about that?
(or make `isLeafHashValid` return an err and use that here?)

Reply:
I didn't want to change the messages, but I agree with you. Changed `isLeafHash() bool` to `validateLeafHash() error` (and tweaked messages a bit as a result of it).